processor/k8sattributesprocessor/options.go

```diff
@@ -178,16 +179,36 @@ func extractFieldRules(fieldType string, fields ...FieldExtractConfig) ([]kube.F
 		}
 
 		var keyRegex *regexp.Regexp
+		var hasKeyRegexReference bool
 		if a.KeyRegex != "" {
 			var err error
 			keyRegex, err = regexp.Compile(a.KeyRegex)
 			if err != nil {
 				return rules, err
 			}
+
+			var maxCapGroup int
+			capGroupRegex := regexp.MustCompile(`\$\d+`)
+			groups := capGroupRegex.FindAllStringSubmatch(a.TagName, -1)
+			for i := 0; i < len(groups); i++ {
+				s := groups[i][0]
+				if c, err := strconv.Atoi(s[1:]); err != nil {
+					return rules, fmt.Errorf("invalid captured group %s: %s", a.TagName, err)
+				} else if c > maxCapGroup {
+					maxCapGroup = c
+				}
+			}
+			subexp := keyRegex.NumSubexp()
+			if maxCapGroup > subexp {
+				return rules, fmt.Errorf("invalid captured group reference")
+			}
+			if subexp > 0 {
+				hasKeyRegexReference = true
+			}
```

Comment:
Looking at how simple it is to do the substitutions, I'm not sure we even need these validation. If we don't do it, we can support named groups as well which is good. references that don't have matching captions will be expanded to empty strings, which is ok I think.

Reply:
yeah sounds good to me. It will be tricky to do validation for both.

Reply:
Ok, if you agree, let's remove this validation block then and document that both named and positioned groups are supported

Reply:
updated doc, the part of defining hasKeyRegexReference has been kept, it is not necessary and can be removed if we have a more elegant way to do substitution for the entire key.